### PR TITLE
Add docs about building and running the trace agent

### PR DESCRIPTION
**Pull request by martpier**
[skip-ci]

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [x] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [x] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [x] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [x] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [x] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [x] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: docs/dev/agent_build.md [hunk 1]
```diff
@@ -45,6 +45,8 @@ This is the complete list of the available components:
 Please note you might need to provide some extra dependencies in your dev
 environment to build certain bits (see [development environment][dev-env]).
 
+Also note that the trace agent needs to be built and run separately: see [here](./trace-agent/READE.md).
```
Comment (Kaderinho) [inline docs/dev/agent_build.md:48]:
```suggestion
Also note that the trace agent needs to be built and run separately: see [here](../trace-agent/README.md).
```

Comment (Kaderinho) [reply docs/dev/agent_build.md:48]: I'm not sure but don't you need `..` before to refer to `docs/trace-agent` folder as you are in the `docs/dev` folder ?

Comment (martpier) [reply docs/dev/agent_build.md]: Thanks, fixed 😳

File: docs/dev/agent_build.md [hunk 2]
```diff
@@ -21,7 +21,7 @@ invoke agent.build --build-exclude=systemd,python
 
 This is the complete list of the available components:
 
-* `apm`: make the APM agent execution available.
+* `apm`: make the APM agent execution available (see [here](../trace-agent/README.md) to build the actual trace agent)
```
Comment (hestonhoffman) [inline docs/dev/agent_build.md:24]:
```suggestion
* `apm`: make the APM agent execution available. For information on building the trace agent, see the trace agent README](../trace-agent/README.md).
```

File: docs/dev/agent_build.md [hunk 3]
```diff
@@ -45,6 +45,8 @@ This is the complete list of the available components:
 Please note you might need to provide some extra dependencies in your dev
 environment to build certain bits (see [development environment][dev-env]).
 
+Also note that the trace agent needs to be built and run separately: see [here](../trace-agent/README.md).
```
Comment (hestonhoffman) [inline docs/dev/agent_build.md:48]:
```suggestion
Also note that the trace agent needs to be built and run separately. For more information, see [the trace agent README](../trace-agent/README.md).
```

**Review by hestonhoffman — APPROVED** [review]

**Pull request merged**